Authentication-Results: mail-a.sr.ht; dkim=pass header.d=gmail.com header.i=@gmail.com
Received: from mail-pl1-f178.google.com (mail-pl1-f178.google.com [209.85.214.178])
	by mail-a.sr.ht (Postfix) with ESMTPS id EB8D820494
	for <~mpu/qbe@lists.sr.ht>; Thu, 09 Jan 2025 23:29:47 +0000 (UTC)
Received: by mail-pl1-f178.google.com with SMTP id d9443c01a7336-21654fdd5daso23412975ad.1
        for <~mpu/qbe@lists.sr.ht>; Thu, 09 Jan 2025 15:29:47 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20230601; t=1736465386; x=1737070186; darn=lists.sr.ht;
        h=content-transfer-encoding:mime-version:message-id:date:subject:cc
         :to:from:from:to:cc:subject:date:message-id:reply-to;
        bh=cIycH907DFB0ROWqOZIy5JJbrTYJJ6JhwNe6e44jX1A=;
        b=MJfHLMmqPf6iskdFDnM3r/S76cnG6lwEh1Vc4tDq9F3vcTNC75oAb91D94yti16Juw
         5I9ZTU5lVDIEr7moegx7iiB7TLQNHWIqbjGKkvUbym7LUzU0qnteuLLIa/nAtp82IbgM
         7OdO+GdsZue3Ono8+W+OIPMWUHBCR0OLIZlexcvZS9mAjTCopCIbOJjxbnGomoZTtcyr
         RYhC6n/27CYYJTCy8drasZdfEA10dZpNe710AYL/iTu6wsZurMldRMi0T2bTu9fUDVQg
         HJe2kpPc4ZmbEGBVq4ZT/A+kEHF8nobX+CHc3nsWUpAsUrJIdmND55HZ03XVoJCr9TwA
         b8tA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1736465386; x=1737070186;
        h=content-transfer-encoding:mime-version:message-id:date:subject:cc
         :to:from:x-gm-message-state:from:to:cc:subject:date:message-id
         :reply-to;
        bh=cIycH907DFB0ROWqOZIy5JJbrTYJJ6JhwNe6e44jX1A=;
        b=tZqVja+6TV03hjTiEWJ29ZFlLp5NT8s0DCcEZaCs/Tlr14wMpno0+RJemirE9p72wR
         QNpeAfsQAW7wHl+g3yrtu6e4Mqqksv9j8Z1J3zMdrPSn3Aq8d1iCxPceq5JMAjAPWwz5
         3ASwqWegJqof3DA3BePPun3yYjOryrgtwfzl8b62qWp65P3MMno2fuUyAqdWG/eElTud
         /u357+2sy0x+JrUhlFduYK0Tje3ahqJJtdjHWkOvm/Stp7TjoNV1sWOi4TszNVxe4zox
         XJU57JBBSHZ6ceVVlVt/RXEfNXc8rN97eo/IH06FGg3qugRjLScsdWn5awpX4Ca4FIA0
         //vg==
X-Gm-Message-State: AOJu0YzWeTMTVfcFXhSU63Ybqf+YsKhJaf8POzK0ouUJMAsf3Deg/Efo
	caE/oyV0EkIHOX/7y96lr4LIqI2hxsZ2cArxvWEi9Ix+vLTvL56BoUTg/w==
X-Gm-Gg: ASbGncvWO8oBJc9UWN2S+8lRYIqXwN3ayPD32IP2QTzkzOxqsPj1AYPoDWQ0OOWix3S
	I0DUc4ePIvdnBYV6j19e+7DzTLKfMDCSO6YRMNZmZrDps51TifuINs3SGtiPXo9sA88YpNbIV3l
	N7A9MThhYOmb63vvGNjAd5+Yv9vrwt1/Ch8TzyD1JJZbE4X3p99f/sMZSA4a6kIwBqlVxk7/ZLP
	4REmFZxomIGY4Z/s8zHTvyOsABd2wsYSY6xC3Iieg+FWsnEFSIhI7v9G2LlHVzsivVM7Br0CBNc
	zJGsAqx6Qjo7UEbI5lKf+1d7LqaP6Nk=
X-Google-Smtp-Source: AGHT+IHXJqo5T34pbmn2lEHu/G/t0tSHgP4qyI8lxmNoC6qxat6AEoyZcEevqCuSpvGTlKKnEXlDqQ==
X-Received: by 2002:a05:6a00:410d:b0:728:b601:86ee with SMTP id d2e1a72fcca58-72d21fd2f0emr11785055b3a.16.1736465386387;
        Thu, 09 Jan 2025 15:29:46 -0800 (PST)
Received: from localhost.localdomain (d154-5-57-197.bchsia.telus.net. [154.5.57.197])
        by smtp.gmail.com with ESMTPSA id d2e1a72fcca58-72d4067e9d4sm343404b3a.127.2025.01.09.15.29.45
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Thu, 09 Jan 2025 15:29:45 -0800 (PST)
From: Scott Graham <sgraham@gmail.com>
To: ~mpu/qbe@lists.sr.ht
Cc: Scott Graham <sgraham@gmail.com>
Subject: [PATCH] fix -Wsometimes-uninitialized warning in non DEBUG builds
Date: Thu,  9 Jan 2025 15:28:55 -0800
Message-ID: <20250109232854.782-2-sgraham@gmail.com>
X-Mailer: git-send-email 2.47.0.vfs.0.3
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit

---
 rv64/isel.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rv64/isel.c b/rv64/isel.c
index 8921a07..99cc158 100644
--- a/rv64/isel.c
+++ b/rv64/isel.c
@@ -154,7 +154,7 @@ selcmp(Ins i, int k, int op, Fn *fn)
 		i.op = KWIDE(k) ? Oceqd : Oceqs;
 		break;
 	default:
-		assert(0 && "unknown comparison");
+		die("unknown comparison");
 	}
 	if (op < NCmpI)
 		i.op = sign ? Ocsltl : Ocultl;
-- 
2.47.0.vfs.0.3

